src/vos/vos_ilog.c

```diff
@@ -215,6 +216,19 @@ vos_parse_ilog(struct vos_ilog_info *info, daos_epoch_t epoch,
 		info->ii_create = entry.ie_id.id_epoch;
 	}
 
+	if (epr->epr_lo != 0) {
```

Comment:
Minor: Maybe check the existence of the "update entries <= epr_lo" in above loop and return immediately if the intention is for "punched check"? And looks "ii_full_scan" isn't necessary, using "ii_create" should be enough.

Reply:
I mean when the intention of this call is 'punched check', 'ii_create' could be used to represent if the key should be removed, so if the this function is to do 'punched check' against a EPR, above loop needs be modified to return 'ii_create = true' when the EPR is above snapshot 1.

But I agree that could make the code more complicated :), it's not a big deal anyway.

Reply:
Ah, I see.  Yeah, it could be done but I think if we go down that road, it may be better to just make it a new call.  That may be better in the long run as it may already be too overloaded.